target/linux/ath79/image/generic.mk

```diff
@@ -448,6 +459,20 @@ define Device/devolo_magic-2-wifi
 endef
 TARGET_DEVICES += devolo_magic-2-wifi
 
+define Device/dlink_dch-g020-a1
+  SOC := qca9531
+  DEVICE_VENDOR := D-Link
+  DEVICE_MODEL := DCH-G020
+  DEVICE_VARIANT := A1
+  DEVICE_PACKAGES := kmod-gpio-pca953x kmod-i2c-gpio kmod-usb2 kmod-usb-acm
+  IMAGES := factory.bin sysupgrade.bin
```

Comment:
I'd have chosen IMAGES += factory.bin here

Reply:
Agreed.
I wonder if there is any way to create the `factory.bin` based on sysupgrade without explicitly specifying the  full command? e.g. `$(IMAGE/sysupgrade.bin) |  mkdapimg2 0x20000`, since the `sysupgrade` also works without giving the full structure.

Reply:
While that would technically work, IMAGE/sysupgrade.bin will insert the OpenWrt-specific metadata into the factory image, which is useless and therefore not used.